source/game_sa/CarAI.cpp

```diff
@@ -55,27 +65,113 @@ void CCarAI::AddAmbulanceOccupants(CVehicle* vehicle) {
+            vehicle->SetUpDriver(PED_TYPE_NONE, false, false);
+            vehicle->SetupPassenger(0, PED_TYPE_NONE, false, false);
+            vehicle->SetupPassenger(1, PED_TYPE_NONE, false, false);
+            vehicle->SetupPassenger(2, PED_TYPE_NONE, false, false);
+            return;
+        case MODEL_PREDATOR:
+            if (FindPlayerPed()->GetWantedLevel() > 1) {
+                CPed* driver = vehicle->SetUpDriver(PED_TYPE_NONE, false, false);
+                CTaskSimpleCarSetPedOut{ vehicle, TARGET_DOOR_DRIVER, true }.ProcessPed(driver);
+                driver->AttachPedToEntity(vehicle, CVector(0.0, 0.0, 0.0), 0, (float)6.2831855, WEAPON_PISTOL);
+                // ??which flag it sets?
+                // driver->m_nPedFlags |= 0x200000u;
+
+                // ??there is no CTaskComplexKillPedFromBoat
+                // driver->GetTaskManager().SetTask(new CTaskComplexKillPedFromBoat{ FindPlayerPed() }, TASK_PRIMARY_PRIMARY);
+            }
+            vehicle->SetUpDriver(PED_TYPE_NONE, false, false);
+        case MODEL_RHINO:
+        case MODEL_COPBIKE:
+            vehicle->SetUpDriver(PED_TYPE_NONE, false, false);
+            return;
+        case MODEL_BARRACKS:
+        case MODEL_COPCARLA:
+        case MODEL_COPCARSF:
+        case MODEL_COPCARVG:
+        case MODEL_COPCARRU:
+            CPed*  driver      = vehicle->SetUpDriver(PED_TYPE_NONE, false, false);
+            uint32 wantedLevel = FindPlayerPed()->GetWantedLevel();
+            if (wantedLevel > 1) {
+                CPed* passenger = vehicle->SetupPassenger(0, PED_TYPE_NONE, false, false);
+                if (wantedLevel > 2) {
+                    CPed* selectedPed;
+                    if (CGeneral::GetRandomNumberInRange(0.0, 1.0) < 0.25) {
+                        selectedPed = driver;
+                    }
+                    if (CGeneral::GetRandomNumberInRange(0.0, 1.0) < 0.25) {
+                        selectedPed = passenger;
+                    }
+                    selectedPed->GiveDelayedWeapon(WEAPON_SHOTGUN, 1'000);
+                }
+                driver->GetIntelligence()->ClearTasks(true, true);
+                driver->GetTaskManager().SetTask(new CTaskComplexCopInCar{ vehicle, passenger, FindPlayerPed(), true }, TASK_PRIMARY_PRIMARY, true);
+                passenger->GetIntelligence()->ClearTasks(true, true);
+                driver->GetTaskManager().SetTask(new CTaskComplexCopInCar{ vehicle, driver, FindPlayerPed(), true }, TASK_PRIMARY_PRIMARY, false);
+                return;
+            }
+
+            if (arg2 || CGeneral::GetRandomNumberInRange(0, 100) < 50) {
+                vehicle->SetupPassenger(0, PED_TYPE_NONE, false, false);
+            }
+        }
+    }
 }
 
 // 0x41BFA0
 void CCarAI::BackToCruisingIfNoWantedLevel(CVehicle* vehicle) {
-    plugin::Call<0x41BFA0, CVehicle*>(vehicle);
+    if (vehicle->vehicleFlags.bIsLawEnforcer) {
+        CWanted* wanted = FindPlayerWanted(-1);
```

Comment:
```suggestion
        CWanted* wanted = FindPlayerWanted();
```

Please consider checking function definitions.